OOModel/src/expressions/MetaCallExpression.h

```diff
@@ -49,6 +52,22 @@ class OOMODEL_API MetaCallExpression: public Super<Expression>
 		 * Returns the meta definition used in this meta call if one exists.
 		 */
 		MetaDefinition* metaDefinition();
+
+		/**
+		 * Generates the tree resulting from this meta call and returns it.
+		 */
+		Declaration* generate();
```

Comment:
Ohh, having two public methods called `generate` and `generated` is a recipe for frustration. We should try to hide stuff that is not generally needed and introduce a single public entry point for generation (where we use `begin/end`).

Either way, having two methods with almost the same name, regardless of whether they are private or public, is not good. Use more descriptive names, e.g. `generate` and `cache`. These names only make sense when private though. The public API should not care about caches and stuff. It should just have one method which fetches the generated code (and internally refreshes the cache if necessary).